src/Model/Response/ReferenceItem.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\CdtClient\Model\Response;
+
+/**
+ * Class ReferenceItem.
+ *
+ * Represents the single item of business reference data obtained from the CDT API.
+ */
+class ReferenceItem
+{
+    /**
+     * The code of the value.
+     */
+    protected string $code;
+
+    /**
+     * The description of the value.
+     */
+    protected string $description;
+
+    /**
+     * @param array<string, mixed> $values
+     */
+    public function __construct(array $values = [])
```

Comment:
This allows you to create a DTO like this

```php
new ReferenceItem([ 'code
  'code' => 'XX',
  'description' => 'YY'
])
```
This way, the BaseCollection will be able to unserialize the object in its constructor.
This is one way to solve the problem with deserializing collections.